modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java

```diff
@@ -0,0 +1,506 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.operations.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.FeatureManagementException;
+import org.wso2.carbon.device.mgt.common.notification.mgt.Notification;
+import org.wso2.carbon.device.mgt.common.notification.mgt.NotificationManagementException;
+import org.wso2.carbon.device.mgt.common.notification.mgt.NotificationManagementService;
+import org.wso2.carbon.device.mgt.common.operation.mgt.Operation;
+import org.wso2.carbon.device.mgt.common.operation.mgt.OperationManagementException;
+import org.wso2.carbon.mdm.mobileservices.windows.common.PluginConstants;
+import org.wso2.carbon.mdm.mobileservices.windows.common.exceptions.WindowsDeviceEnrolmentException;
+import org.wso2.carbon.mdm.mobileservices.windows.common.util.WindowsAPIUtils;
+import org.wso2.carbon.mdm.mobileservices.windows.operations.*;
+import org.wso2.carbon.mdm.mobileservices.windows.services.syncml.beans.Profile;
+import org.wso2.carbon.mdm.mobileservices.windows.services.syncml.util.SyncmlUtils;
+import org.wso2.carbon.policy.mgt.common.ProfileFeature;
+import org.wso2.carbon.policy.mgt.common.monitor.ComplianceFeature;
+import org.wso2.carbon.policy.mgt.common.monitor.PolicyComplianceException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.mdm.mobileservices.windows.common.util.WindowsAPIUtils.convertToDeviceIdentifierObject;
+
+/**
+ * Class contains Operation related utilities.
+ */
+public class OperationUtils {
+    private static Log log = LogFactory.getLog(OperationUtils.class);
+    List<? extends Operation> pendingDataOperations;
+
+
+    /**
+     * Update the operations using device status payload.
+     *
+     * @param status           Client side status for the specific operations
+     * @param syncmlDocument   syncml payload for operation status which parse through  the syncml engine
+     * @param deviceIdentifier specific device identifier for each device
+     * @throws OperationManagementException
+     * @throws DeviceManagementException
+     */
+    public void updateDeviceOperations(Status status, SyncmlDocument syncmlDocument,
+                                       DeviceIdentifier deviceIdentifier)
+            throws OperationManagementException, DeviceManagementException, NotificationManagementException,
+            WindowsOperationException {
+
+        pendingDataOperations = SyncmlUtils.getDeviceManagementService()
+                .getOperationsByDeviceAndStatus(deviceIdentifier, Operation.Status.PENDING);
+        if (status.getData().equals(Constants.SyncMLResponseCodes.ACCEPTED) || status.getData().equals
```

Comment:
As a best practice, always compare the non-deterministic variable values against known constants.